SecurityManager/Core.UnitTests/Persistence/SecurityManagerSqlStorageObjectFactoryTest.cs

```diff
@@ -80,10 +80,11 @@ public void CreateReadOnlyStorageProvider ()
     {
       var result = _securityManagerSqlStorageObjectFactory.CreateReadOnlyStorageProvider(_rdbmsProviderDefinition, _persistenceExtensionStub.Object);
 
-      Assert.That(result, Is.TypeOf(typeof(SecurityManagerRdbmsProvider)));
-      Assert.That(result.As<SecurityManagerRdbmsProvider>().PersistenceExtension, Is.SameAs(_persistenceExtensionStub.Object));
-      Assert.That(result.As<SecurityManagerRdbmsProvider>().StorageProviderDefinition, Is.SameAs(_rdbmsProviderDefinition));
-      Assert.That(result.As<SecurityManagerRdbmsProvider>().ConnectionString, Is.EqualTo(_rdbmsProviderDefinition.ReadOnlyConnectionString));
+      Assert.That(result, Is.TypeOf(typeof(ReadOnlyStorageProviderDecorator)));
```

Comment:
```suggestion
      Assert.That(result, Is.InstanceOf<ReadOnlyStorageProviderDecorator>());
```
I prefer the generic syntax, shorter :)
as for TypeOf vs InstanceOf: one tests exact match, the other allows derived types. Occasionally, it's relevant, other times it's a question of which is used more often.